pytential/symbolic/execution.py

```diff
@@ -309,6 +309,24 @@ def map_common_subexpression(self, expr):
 
     # }}}
 
+    def map_error_expression(self, expr):
+        raise RuntimeError(expr.message)
+
+    def map_is_shape_class(self, expr):
+        discr = self.places.get_discretization(
+            expr.dofdesc.geometry, expr.dofdesc.discr_stage)
+
+        from pytools import is_single_valued
+        if not is_single_valued(type(grp.mesh_el_group) for grp in discr.groups):
+            raise TypeError("non-homogeneous element groups are not supported")
```

Comment:
```suggestion
            # FIXME
            raise NotImplementedError("non-homogeneous element groups are not supported")
```
Conceivably, one could stick `True/False`s into a `DOFArray`.